File: StudentManagementSystem.Frontend/Services/Api/AuthService.cs
using StudentManagementSystem.Frontend.Models.DTOs;
using Blazored.LocalStorage;

namespace StudentManagementSystem.Frontend.Services.Api;

public class AuthService
{
    private readonly ApiService _apiService;
    private readonly ILocalStorageService _localStorage;

    public AuthService(ApiService apiService, ILocalStorageService localStorage)
    {
        _apiService = apiService;
        _localStorage = localStorage;
    }

    public async Task<AuthResponse?> LoginAsync(LoginRequest request)
    {
        var response = await _apiService.PostWithResponseAsync<AuthResponse>("api/auth/login", request);
        
        if (response?.Success == true && response.Data != null)
        {
            await _localStorage.SetItemAsync("authToken", response.Data.Token);
            await _localStorage.SetItemAsync("user", response.Data.User);
            await _localStorage.SetItemAsync("userRole", response.Data.Role);
        }
        
        return response?.Data;
    }

    public async Task<AuthResponse?> RegisterAsync(RegisterRequest request)
    {
        var response = await _apiService.PostWithResponseAsync<AuthResponse>("api/auth/register", request);
        
        if (response?.Success == true && response.Data != null)
        {
            await _localStorage.SetItemAsync("authToken", response.Data.Token);
            await _localStorage.SetItemAsync("user", response.Data.User);
            await _localStorage.SetItemAsync("userRole", response.Data.Role);
        }
        
        return response?.Data;
    }

    public async Task LogoutAsync()
    {
        await _localStorage.RemoveItemAsync("authToken");
        await _localStorage.RemoveItemAsync("user");
        await _localStorage.RemoveItemAsync("userRole");
    }

    public async Task<bool> IsAuthenticatedAsync()
    {
        var token = await _localStorage.GetItemAsync<string>("authToken");
        return !string.IsNullOrEmpty(token);
    }

    public async Task<string?> GetUserRoleAsync()
    {
        return await _localStorage.GetItemAsync<string>("userRole");
    }

    public async Task<UserDto?> GetCurrentUserAsync()
    {
        return await _localStorage.GetItemAsync<UserDto>("user");
    }

    public async Task<string?> GetTokenAsync()
    {
        return await _localStorage.GetItemAsync<string>("authToken");
    }
}

File: frontend/services/api/authService.ts
// services/api/authService.ts
import apiClient from '../../utils/apiClient';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  user: {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    role: string;
  };
}

export interface RegisterRequest {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  role: string;
}

export interface RegisterResponse {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  role: string;
  createdAt: string;
}

class AuthService {
  async login(data: LoginRequest): Promise<LoginResponse> {
    return apiClient.post<LoginResponse>('/auth/login', data);
  }

  async register(data: RegisterRequest): Promise<RegisterResponse> {
    return apiClient.post<RegisterResponse>('/auth/register', data);
  }

  async getCurrentUser() {
    return apiClient.get<LoginResponse['user']>('/auth/me');
  }
}

const authService = new AuthService();
export default authService;